src/back-end/test/createUser.test.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Tests export user creation functions
+ */
+
+/* Mock envvars */
+process.env.HASHKEY = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+
+/* Imports */
+const { createUser } = require("../createFiles/createUser");
+const schema = require("../schema.js");
+const mongoose = require("mongoose");
+
+describe("createUser Tests", () => {
+
+    /**
+     * Checks for an existing user
+     *
+     * @param {Object} userData Object containing email and password
+     * @return true if user exists, false otherwise
+     */
+    const checkUser = async (userData) => {
+        try {
+            const user = await schema.User.findOne({ email: userData.email});
+            if (user === null) {
+                return false;
+            }
+        } catch (err) {
+            return false;
+        }
+    };
+
+    /**
+     * Returns id of user
+     *
+     * @param {String} email The email of the user to find
+     * @returns id of user
+     */
+    const getID = async (email) => {
+        const user = await schema.User.findOne({email: email});
+        return user.signifiers[0].id;
+    }
+
+    /* Connect to the in-memory Mongo server */
+    beforeAll(async () => {
+        mongoose.set("useCreateIndex", true);
+        await mongoose.connect(`${globalThis.__MONGO_URI__}${globalThis.__MONGO_DB_NAME__}`, {
+            useUnifiedTopology: true,
+            useNewUrlParser: true
+        });
+    });
+
+    /* Drop the database */
+    afterEach(async () => {
+        await schema.User.deleteMany({});
+    });
+
+    /* Clean up the connection */
+    afterAll(async () => {
+        await mongoose.connection.close();
+    });
+
+    test("createUser() produces expected user", async () => {
+        const EMAIL = "foo@gmail.com";
+        const PWDHASH = "123ABC";
+        const KEY = "ABC123";
+        const user = {
+            email: EMAIL,
+            pwd: PWDHASH
+        };
+        await createUser(EMAIL, PWDHASH, KEY, () => {
+            let success = checkUser(user);
+            expect(success).toBe(true);
+        });
+    });
+
+    test("makeid() produces unique IDs", async () => {
+        const EMAIL1 = "foo@gmail.com";
+        const EMAIL2 = "faa@gmail.com";
+        const PWDHASH1 = "123ABC";
+        const PWDHASH2 = "456DEF";
+        const KEY1 = "ABC123";
+        const KEY2 = "ABC123";
+        await createUser(EMAIL1, PWDHASH1, KEY1, () => {
```

Comment:
Same problem, this doesn't test anything because this function doesn't take callbacks.